src/calchartdoc.h

```diff
@@ -165,6 +172,8 @@ class CalChartDoc : public wxDocument
 	bool AlreadyHasPrintContinuity() const;
 	int PrintToPS(std::ostream& buffer, bool eps, bool overview, int min_yards, const std::set<size_t>& isPicked, const CalChartConfiguration& config_) const;
 	
+	void overwriteMusicScore(const MusicScoreDocComponent& newScore);
+	MusicScoreDocComponent* getMusicScore();
```

Comment:
In generally I wouldn't just always return pointers to internal components.  You will find you quickly run into ownership issues (who owns that resource?) or dangling pointers (some code keeps a pointer to the internals and then the object goes away).  Also, you break encapsulation; why go to all the work of making something private if you just give it away?

Also, why a pointer?  Why not a reference?  A pointer is there to imply that the value could be set or unset, meaning it is valid that the Show may not have a Score (may be null).  If that is not true (if an invariant is that all shows MUST have a score), then change to a reference (and also make it const):
const MusicScoreDocComponent& getMusicScore() const;

if the intent is to give the internals for modification, add a non-const overload:
MusicScoreDocComponent& getMusicScore();

So then when clients are calling getMusicScore on a const object, they get a const score, which is natural (if the object is const, then the internals of the object should also be const) and when they have a non-const object, they get non-const internals.